src/test/java/seedu/address/model/transaction/AmountContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.model.transaction;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import seedu.address.testutil.TransactionBuilder;
+
+public class AmountContainsKeywordsPredicateTest {
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("1.33");
+        List<String> secondPredicateKeywordList = Arrays.asList("1.54", "1.22");
+
+        AmountContainsKeywordsPredicate firstPredicate =
+                new AmountContainsKeywordsPredicate(firstPredicateKeywordList);
+        AmountContainsKeywordsPredicate secondPredicate =
+                new AmountContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        AmountContainsKeywordsPredicate firstPredicateCopy =
+                new AmountContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+
+    @Test
+    public void test_amountContainsKeywordsExpenses_returnsTrue() {
+        // One keyword
+        AmountContainsKeywordsPredicate predicate =
+                new AmountContainsKeywordsPredicate(Collections.singletonList("1.22"));
+        assertTrue(predicate.test(new TransactionBuilder().withAmount("1.56").buildExpense()));
+
+        // Multiple keywords
+        predicate = new AmountContainsKeywordsPredicate(Arrays.asList("1", "1.35"));
+        assertTrue(predicate.test(new TransactionBuilder().withAmount("1.33").buildExpense()));
+
+        // Only one matching keyword
+        predicate = new AmountContainsKeywordsPredicate(Arrays.asList("1.2", "9.11"));
+        assertTrue(predicate.test(new TransactionBuilder().withAmount("1.55").buildExpense()));
```

Comment:
wait why will this be true? imma little confused

Reply:
I initially implemented such that 1.2 will result in a range from 1 to 1.99 displayed, but now I changed it alr to only integers can be keyed in to prevent confusion